### PR TITLE
Add Pod as an owner of a CiliumEndpoint and remove useless Delete

### PR DESCRIPTION
**Pull request by aanm**
Remove Delete operation when starting up Cilium which removes the CiliumEndpoint in the k8s apiserver. This is unnecessary since CiliumEndpoint will be updated immediately afterwards.
As a CiliumEndpoint can't exist without a Pod, we should set the Pod as an Owner of a particular endpoint.
This will allow CiliumEndpoints from being automatically GCed by K8s once the Pod is gone.

```release-nite
Do not delete CiliumEndpoint on Cilium restart and add Pod as a Owner of a CiliumEndpoint
```

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: daemon/cmd/endpoint.go [hunk 1]
```diff
@@ -151,24 +152,24 @@ func NewPutEndpointIDHandler(d *Daemon) PutEndpointIDHandler {
 
 // fetchK8sLabelsAndAnnotations wraps the k8s package to fetch and provide
 // endpoint metadata. It implements endpoint.MetadataResolverCB.
-func (d *Daemon) fetchK8sLabelsAndAnnotations(nsName, podName string) (labels.Labels, labels.Labels, map[string]string, error) {
+func (d *Daemon) fetchK8sLabelsAndAnnotations(nsName, podName string) (*types.Pod, labels.Labels, labels.Labels, map[string]string, error) {
```
Comment (tgraf) [inline daemon/cmd/endpoint.go:157]: Is returned pod deep copied? Can you add a comment to the function description saying whether it is or not?

**Review by tgraf — CHANGES_REQUESTED** [review]

File: daemon/cmd/endpoint.go (hunk 1 above)
Comment (aanm) [reply daemon/cmd/endpoint.go:157]: It is, comment added.

**Comment by aanm** [comment]
test-me-please

File: pkg/k8s/watchers/endpointsynchronizer.go [hunk 2]
```diff
@@ -130,34 +130,37 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 						return nil
 					}
 
-					scopedLog.Debug("Deleting CEP during an initialization")
-					err := ciliumClient.CiliumEndpoints(namespace).Delete(ctx, podName, meta_v1.DeleteOptions{})
```
Comment (tgraf) [inline pkg/k8s/watchers/endpointsynchronizer.go:134]: I think this was done a long time ago when we changed the format of the CIliumEndpoint at the very beginning for this feature and this allowed to up and downgrade.

File: pkg/endpoint/endpoint.go [hunk 3]
```diff
@@ -211,6 +212,9 @@ type Endpoint struct {
 	// K8sNamespace is the Kubernetes namespace of the endpoint
 	K8sNamespace string
 
+	// pod
```
Comment (tgraf) [inline pkg/endpoint/endpoint.go:215]: 💎

**Review by tgraf — APPROVED** [review]

**Pull request merged**